metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/api/internal/SearchTermContext.java

```diff
@@ -17,27 +19,11 @@ public SearchTermContext(String textValue, String language, Set<FieldType> field
   }
 
   @Override
-  public List<EntityType> getCandidateTypes() {
-    return fieldTypes.stream().map(FieldType::getEntityType).collect(Collectors.toList());
+  public Set<EntityType> getCandidateTypes() {
+    return fieldTypes.stream().map(FieldType::getEntityType).collect(Collectors.toSet());
   }
 
   public Set<FieldType> getFieldTypes() {
-    return new HashSet<>(fieldTypes);
+    return Collections.unmodifiableSet(fieldTypes);
   }
-
-  @Override
```

Comment:
Same here 

"I think equals and hashcode should be declared here too, preferably with using the new fields or just by using super"